File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem40_MaximumLevelSumOfABinaryTreeTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem40_MaximumLevelSumOfABinaryTreeTest {
    private final Problem40_MaximumLevelSumOfABinaryTree test = new Problem40_MaximumLevelSumOfABinaryTree();

    @Test
    void maxLevelSum() {
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(7);
        root.left.left = new TreeNode(7);
        root.left.right = new TreeNode(-8);
        root.right = new TreeNode(0);

        assertEquals(2, test.maxLevelSum(root));
    }
}

File: JavaScript/leetcode75/src/test/Problem40_MaximumLevelSumOfABinaryTree.test.js
const maxLevelSum = require("../main/Problem40_MaximumLevelSumOfABinaryTree");

//  Definition for a binary tree node.
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

test("should return max level at which sum is greater", () => {
  let root = new TreeNode(1);
  root.left = new TreeNode(7);
  root.left.left = new TreeNode(7);
  root.left.right = new TreeNode(-8);
  root.right = new TreeNode(0);
  expect(maxLevelSum(root)).toBe(2);
});
